packages/server-common/src/handlers/webhooks/contentful.handler.ts

```diff
@@ -25,44 +29,86 @@ const getDetailFromRequest = (
   resourceId: request.payload.sys.id,
   ...request.payload,
 });
+type Config = {
+  eventBus: string;
+  eventSource: string;
+  space: string;
+  environment: string;
+  previewAccessToken: string;
+};
 export const contentfulHandlerFactory =
   (
     webhookAuthenticationToken: string,
     eventBridge: EventBridge,
-    eventBus: string,
-    eventSource: string,
+    config: Config,
     logger: Logger,
   ): ((
     request: lambda.Request<ContentfulWebhookPayload>,
   ) => Promise<{ statusCode: number }>) =>
   async (request) => {
     validateContentfulRequest(request, webhookAuthenticationToken);
 
+    const detailType = getDetailTypeFromRequest(request);
+    const detail = getDetailFromRequest(request);
+
+    if (!detail.sys.revision) {
+      throw new Error('Invalid payload');
+    }
+
+    const entryVersion = detail.sys.revision;
+    const cpaClient = getCPAClient({
+      previewAccessToken: config.previewAccessToken,
+      space: config.space,
+      environment: config.environment,
+    });
+    const fetchEntryById = async () => {
+      try {
+        const entry = await cpaClient.getEntry(detail.resourceId);
+        return entry;
+      } catch (error) {
+        if (
+          error instanceof Error &&
+          error.message.match(/The resource could not be found/)
```

Comment:
We might also want to check that the event being handled is an `Unpublish` before continuing here.